Calibration/HcalAlCaRecoProducers/src/AlCaGammaJetProducer.cc

```diff
@@ -135,13 +135,24 @@ void AlCaGammaJetProducer::endJob() {
   edm::LogInfo("AlcaGammaJet") << "Accepts " << nSelect_ << " events from a total of " << nAll_ << " events";
 }
 
-bool AlCaGammaJetProducer::select (reco::PhotonCollection ph, reco::PFJetCollection jt) {
- 
-  if (jt.size()<1) return false;
-  if (ph.size()<1) return false;
-  if (((jt.at(0)).pt())<minPtJet_)    return false;
-  if (((ph.at(0)).pt())<minPtPhoton_) return false;
-  return true;
+bool AlCaGammaJetProducer::select (const reco::PhotonCollection &ph, const reco::PFJetCollection &jt) {
+
+  // Check the requirement for minimum pT
+  if (ph.size() == 0) return false;
+  bool ok(false);
+  for (reco::PFJetCollection::const_iterator itr=jt.begin();
+       itr!=jt.end(); ++itr) {
+    if (itr->pt() >= minPtJet_) {
+      ok = true;
+      break;
+    }
+  }
+  if (!ok) return ok;
+  for (reco::PhotonCollection::const_iterator itr=ph.begin();
+       itr!=ph.end(); ++itr) {
+    if (itr->pt() >= minPtPhoton_) return ok;
+  }
+  return false;
 }
```

Comment:
Yes - we need events with at least one high pt jet and one high pt photon. The assumption of ordering was found to be incorrect - so the code is optimized this way.
________________________________
From: Salvatore Di Guida [notifications@github.com]
Sent: 13 May 2015 10:17
To: cms-sw/cmssw
Cc: Sunanda Banerjee
Subject: Re: [cmssw] bsunanda:Run2 alca13 Get rid of unused variables and avoid explicit names (#9062)


In Calibration/HcalAlCaRecoProducers/src/AlCaGammaJetProducer.cc<https://github.com/cms-sw/cmssw/pull/9062#discussion_r30210248>:

> +  // Check the requirement for minimum pT
> +  if (ph.size() == 0) return false;
> +  bool ok(false);
> +  for (reco::PFJetCollection::const_iterator itr=jt.begin();
> +       itr!=jt.end(); ++itr) {
> +    if (itr->pt() >= minPtJet_) {
> +      ok = true;
> +      break;
> +    }
> +  }
> +  if (!ok) return ok;
> +  for (reco::PhotonCollection::const_iterator itr=ph.begin();
> +       itr!=ph.end(); ++itr) {
> +    if (itr->pt() >= minPtPhoton_) return ok;
> +  }
> +  return false;
>  }


@bsunanda<https://github.com/bsunanda> IIUC the tests you perform are:

  1.  Test if the Photon collection is empty: return false if so;
  2.  Check the requirement for minimum pT: at least one PFJet should have pt larger than the threshold, otherwise return false;
  3.  Check the requirement for minimum pT: at least one Photon should have pt larger than the threshold. If so, return true, otherwise false.

—
Reply to this email directly or view it on GitHub<https://github.com/cms-sw/cmssw/pull/9062/files#r30210248>.